lib/ecto/query.ex

```diff
@@ -1806,8 +1814,8 @@ defmodule Ecto.Query do
       City |> order_by(^order_by_param) # Keyword list
 
   """
-  defmacro order_by(query, binding \\ [], expr) do
-    Builder.OrderBy.build(query, binding, expr, __CALLER__)
+  defmacro order_by(query, binding \\ [], expr, opts \\ []) do
+    Builder.OrderBy.build(query, binding, expr, opts[:position], __CALLER__)
```

Comment:
We cannot access it here because it means we can't do this:

    opts = [position: :prepend]
    order_by(query, [p], p.value, opts)

I think we should either allow dynamic options (and reading it only at runtime) or require an atom to be given (which can be compile-time only).

Reply:
Thanks, I updated so it's evaluated at runtime. Does it seem alright?